SNOW-1435047 fix insert with binding data on different paramstyle (#1629)
snowflakedb/snowpark-python: CHANGELOG.md

- Improved error message to remind users set `{"infer_schema": True}` when reading csv file without specifying its schema.

### Bug Fixes

> - Fixed a bug where `Session.create_dataframe` is called with more than 512 rows throws `TypeError` when creating a session with `format` or `pyformat` `paramstyle`.

Reviewer: just reading the code, why it's the number `512` that triggers the error?

Author: 512 is set as `analyzer.ARRAY_BIND_THRESHOLD`

> - Fixed a bug where `Session.create_dataframe` when created with large greater than 512 rows threw `TypeError` when created a session with `format` or `pyformat` `paramstyle`.

Reviewer 2: ```suggestion
- Fixed a bug where `Session.create_dataframe` is called with more than 512 rows throws `TypeError` when creating a session with `format` or `pyformat` `paramstyle`.
```
